clean_air/util/file_converter.py

```diff
@@ -0,0 +1,174 @@
+        form_data = {'title': row[1].values[16],
+                     'description': row[1].values[17],
+                     'firstname1': row[1].values[6],
+                     'surname1': row[1].Surname,
+                     'firstname2': row[1].values[11],
+                     'surname2': row[1].values[12],
+                     'north': row[1].values[42],
+                     'south': row[1].values[41],
+                     'east': row[1].values[43],
+                     'west': row[1].values[44],
+                     'chemicals': row[1].values[46].split(';'),
+                     'obs_level': row[1].values[19],
+                     'data_source': row[1].values[20],
+                     'time_range_start': row[1].values[37],
+                     'time_range_end': row[1].values[38],
+                     'lineage': row[1].values[50],
+                     'quality': row[1].values[51],
+                     'docs': row[1].values[22]}
+
+        form = pd.Series(data=form_data)
+        form_responses.append([form, r])
+
+    return form_responses
+
+
+def save_as_json(data_object, r, output_location):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as JSON string.
+    """
+    # Pollutants(chemicals) must be organised into a useable structure before
+    # being entered into the new file:
+    chemicals = []
+    for chem in data_object.chemicals:
+        if chem != "":
+            chems = {'name': chem,
+                     'shortname': chem[chem.find("(") + 1:chem.find(")")],
+                     'chart': "url/to/chart/image.(png|jpg|svg|etc)"}
+            chemicals.append(chems)
+
+    new_file = {"pollutants": chemicals,
+                "environmentType": data_object.obs_level,
+                "dateRange": {"startDate": data_object.time_range_start,
+                              "endDate": data_object.time_range_end}}
+
+    # write the dictionary above (new_file) to a json in the cap-sample-data
+    # directory with the addition of a unit to indicate the number of the
+    # metadata form response.
+    with open(output_location + str(r) + '.json', 'w') as fp:
+        json.dump(new_file, fp, indent=2, cls=DateTimeEncoder)
+
+
+def save_as_yaml(data_object, r, output_location):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as yaml.
+    """
+    # First extract the shortname only for chemical species:
+    chem_species = []
+    for chem in data_object.chemicals:
+        if chem != "":
+            chem_shortname = chem[chem.find("(") + 1:chem.find(")")]
+            # If we leave chemical species in this format something (yaml?)
+            # automatically adds inverticommas to 'NO' but nothing else, so
+            # add parentheses back on to standardise format for yaml:
+            chem_shortname = chem_shortname.replace(chem_shortname,
+                                                    "(" + chem_shortname + ")")
```

Comment:
need to discuss this.  not really happy with yaml output converting 'NO' in a different way to all other chemicals.

Reply:
Perhaps there is an option to `yaml.dump` to always quote strings then?

I agree it would be a bit strange to see only one quoted string in the output. For the most part it's not expected that anyone would look at the file though, it just needs to be parsed correctly by the metadata handler, which it would be.

Reply:
we discussed this in person and i agreed to just ignore the way the yaml output looks.  this is not a problem that needed solving so i have removed the bit that solves it.  does seem more straightforward now.